xarray/tests/test_backends.py

```diff
@@ -2172,14 +2172,18 @@ def create_store(self):
             yield backends.H5NetCDFStore(tmp_file, "w")
 
     @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
-    def test_complex(self):
+    def test_complex_warning(self):
         expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
         with pytest.warns(FutureWarning):
-            # TODO: make it possible to write invalid netCDF files from xarray
-            # without a warning
             with self.roundtrip(expected) as actual:
                 assert_equal(expected, actual)
 
+    def test_complex(self):
+        expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
+        save_kwargs = {"invalid_netcdf": True}
+        with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
```

Comment:
Let's add an explicit test that there is no warning here: https://stackoverflow.com/questions/45671803/how-to-use-pytest-to-assert-no-warning-is-raised

Reply:
Yep, did it.